bin/prover-client/src/checkpoint_runner/runner.rs

```diff
@@ -0,0 +1,65 @@
+use std::sync::Arc;
+
+use strata_rocksdb::prover::db::ProofDb;
+use tokio::{
+    sync::Mutex,
+    time::{interval, Duration},
+};
+use tracing::{error, info};
+
+use crate::{
+    checkpoint_runner::fetch::fetch_latest_checkpoint_index,
+    operators::{checkpoint::CheckpointOperator, ProvingOp},
+    task_tracker::TaskTracker,
+};
+
+const CHECKPOINT_POLL_INTERVAL: u64 = 10;
+
+/// Periodically polls for the latest checkpoint index and updates the current index.
+/// Dispatches tasks when a new checkpoint is detected.
+pub async fn checkpoint_proof_runner(
+    operator: CheckpointOperator,
+    task_tracker: Arc<Mutex<TaskTracker>>,
+    db: Arc<ProofDb>,
+) {
+    info!("Checkpoint runner started");
+    let mut ticker = interval(Duration::from_secs(CHECKPOINT_POLL_INTERVAL));
+    let mut current_checkpoint_idx: Option<u64> = None;
+
+    loop {
+        ticker.tick().await;
+
+        if let Err(e) =
+            process_checkpoint(&operator, &task_tracker, &db, &mut current_checkpoint_idx).await
+        {
+            error!("Error processing checkpoint: {e:?}");
```

Comment:
Use log params instead of interpolation: `error!(%checkpoint_idx, err = ?e, "error processing checkpoint");`